src/test/java/org/folio/search/controller/SearchInstanceIT.java

```diff
@@ -2,119 +2,63 @@
-  }
-
-  @Test
-  void canSearchByContributorsName() throws Exception {
-    mockMvc.perform(get(searchInstancesByQuery("contributors.name all {value}"), "frank")
-      .headers(defaultHeaders()))
-      .andExpect(status().isOk())
-      .andExpect(jsonPath("totalRecords", is(1)))
-      .andExpect(jsonPath("instances[0].id",
-        is(getSemanticWeb().getId())))
-      .andExpect(jsonPath("instances[0].contributors[0].name",
-        is("Antoniou, Grigoris")))
-      .andExpect(jsonPath("instances[0].contributors[1].name",
-        is("Van Harmelen, Frank")));
-  }
-
-  @Test
-  void canSearchByContributors() throws Exception {
-    mockMvc.perform(get(searchInstancesByQuery("contributors all {value}"), "grigoris")
-      .headers(defaultHeaders()))
-      .andExpect(status().isOk())
-      .andExpect(jsonPath("totalRecords", is(1)))
-      .andExpect(jsonPath("instances[0].id", is(getSemanticWeb().getId())));
+  private static Stream<Arguments> positiveSearchTestDataProvider() {
+    return Stream.of(
+      arguments("search by instance id for exactMatch", "id={value}", array(getSemanticWeb().getId()), null),
+      arguments("search by instance id using wildcard", "id={value}", array("5bf370e0*a0a39"), null),
+      arguments("search by instance title (title)", "title all {value}", array("semantic"), null),
+      arguments("search by instance title (series)", "title all {value}", array("cooperative"), null),
+      arguments("search by instance title (series, partial)", "title all {value}", array("cooperate"), null),
+      arguments("search by instance title (part of title)", "title all {value}", array("primers"), null),
+      arguments("search by instance title (alternative title)", "title all {value}", array("primers"), null),
+      arguments("search by publisher (abbreviate)", "publisher all {value}", array("MIT"), null),
+      arguments("search by publisher (abbreviate lowercase)", "publisher all {value}", array("mit"), null),
+      arguments("search by publisher (word)", "publisher all {value}", array("press"), null),
+      arguments("search by contributors name", "contributors.name all {value}", array("frank"),
+        (ThrowingConsumer<ResultActions>) searchResult -> searchResult
+          .andExpect(jsonPath("instances[0].contributors[0].name", is("Antoniou, Grigoris")))
+          .andExpect(jsonPath("instances[0].contributors[1].name", is("Van Harmelen, Frank")))),
+      arguments("search by contributors alias", "contributors all {value}", array("grigoris"), null),
+      arguments("search by hrid for exact match", "hrid={value}", array("inst000000000022"), null),
+      arguments("search by hrid with wildcard (starts with)", "hrid={value}", array("inst000*"), null),
+      arguments("search by hrid with wildcard (ends with)", "hrid={value}", array("*00022"), null),
+      arguments("search by hrid with wildcard (contains)", "hrid={value}", array("*00000002*"), null)
```

Comment:
the 3rd argument is array, but all the cases requires only one argument. Can we make it a single value?

Reply:
yes, but if we will implement bool query with 2 arguments, we have to refactor this again